tendermint/src/lite/types.rs

```diff
@@ -87,19 +87,21 @@ pub trait TrustThreshold: Copy + Clone {
 /// voting power signed (in other words at least one honest validator signed).
 /// Some clients might require more than +1/3 and can implement their own
 /// [`TrustThreshold`] which can be passed into all relevant methods.
-#[derive(Copy, Clone)]
+#[derive(Copy, Clone, Debug, PartialEq)]
 pub struct TrustThresholdFraction {
     numerator: u64,
     denominator: u64,
 }
 
 impl TrustThresholdFraction {
-    pub fn new(numerator: u64, denominator: u64) -> Self {
-        // TODO: should we make sure we got a proper fraction here?
-        Self {
-            numerator,
-            denominator,
+    pub fn new(numerator: u64, denominator: u64) -> Result<Self, Error> {
+        if numerator <= denominator && denominator > 0 {
```

Comment:
Actually I wonder if we should bound this below at 1/3 ... it's not really safe with anything less than that